lib/cdo/app_server_hooks.rb

```diff
@@ -7,30 +7,24 @@ def self.before_fork
       DASHBOARD_DB.disconnect
       Cdo::AppServerMetrics.instance&.spawn_reporting_task if defined?(Cdo::AppServerMetrics)
 
-      # Control automated restarts of web application server processes via Gatekeeper.
+      # Control automated restarts of web application server processes via Gatekeeper and DCDO.
       # NOTE: before_fork runs on the parent puma process, so complete restart of the web application services on all
-      # front end instances is required for a change of this Gatekeeper flag to take effect:
+      # front end instances is required for a change of these Gatekeeper and DCDO flags to take effect:
       #   sudo systemctl restart dashboard && sudo systemctl restart pegasus
       require 'dynamic_config/gatekeeper'
       require 'dynamic_config/dcdo'
-      Gatekeeper.after_fork
-      DCDO.after_fork
 
       if Gatekeeper.allows('enableWebServiceProcessRollingRestart')
         require 'puma_worker_killer'
 
-        restart_period = DCDO.get("web_service_process_restart_period", 12 * 3600) # default to 12 hours
+        restart_period = DCDO.get('web_service_process_restart_period', 12 * 3600) # default to 12 hours
         PumaWorkerKiller.enable_rolling_restart(restart_period)
       end
     end
 
     def self.after_fork(host:)
       require 'cdo/aws/metrics'
       Cdo::Metrics.put('App Server/WorkerBoot', 1, Host: host)
-      require 'dynamic_config/gatekeeper'
-      require 'dynamic_config/dcdo'
-      Gatekeeper.after_fork
-      DCDO.after_fork
```

Comment:
Really nice to see a reduction in `after_fork` games we have to play.

Reply:
Right?! I'm not even that upset that this threading stuff broke unexpectedly, given how many hoops we had to jump through to keep it working :upside_down_face: